azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -98,12 +115,85 @@ public ExecutorManager(Props props, ExecutorLoader loader,
     executingManager = new ExecutingManagerUpdaterThread();
     executingManager.start();
 
+    queueProcessor = new QueueProcessorThread();
+    queueProcessor.start();
+
     long executionLogsRetentionMs =
-        props.getLong("execution.logs.retention.ms",
-            DEFAULT_EXECUTION_LOGS_RETENTION_MS);
+      props.getLong("execution.logs.retention.ms",
+        DEFAULT_EXECUTION_LOGS_RETENTION_MS);
     cleanerThread = new CleanerThread(executionLogsRetentionMs);
     cleanerThread.start();
+  }
 
+  /**
+   * <pre>
+   * Setup activeExecutors using azkaban.properties and database executors
+   * Note:
+   * 1. If a local executor is specified and it is missing from db,
+   *    this method add local executor as active in DB
+   * 2. If a local executor is specified and it is marked inactive in db,
+   *    this method will convert local executor as active in DB
+   * 3. If azkaban.use.multiple.executors is set true, this method will
+   *    load all active projects
+   * </pre>
+   *
+   * @throws ExecutorManagerException
+   */
+  public void setupExecutors() throws ExecutorManagerException {
```

Comment:
simplified use case to assume that we will not have local as well as multi-executor simultaneously. Though we need to make a db entry due to the way we are now defining a flow running i.e. it has an assigned executor. So in local mode, we need to have an executor to assign to. This db requirement is for the use case where we use the ability to bounce webserver without affecting running flows.